client/reader/post-byline/index.jsx

```diff
@@ -1,40 +1,63 @@
 /**
  * External Dependencies
  */
-var React = require( 'react' );
+import React from 'react';
 
 /**
  * Internal Dependencies
  */
-var ExternalLink = require( 'components/external-link' ),
-	Gravatar = require( 'components/gravatar' ),
-	Gridicon = require( 'components/gridicon' ),
-	PostTime = require( 'reader/post-time' ),
-	utils = require( 'reader/utils' ),
-	stats = require( 'reader/stats' );
-
-var PostByline = React.createClass( {
-
-	recordTagClick: function() {
-		stats.recordAction( 'click_tag' );
-		stats.recordGaEvent( 'Clicked Tag Link' );
-		stats.recordTrackForPost( 'calypso_reader_tag_clicked', this.props.post, {
+import ExternalLink from 'components/external-link';
+import Gravatar from 'components/gravatar';
+import Gridicon from 'components/gridicon';
+import PostTime from 'reader/post-time';
+import { siteNameFromSiteAndPost } from 'reader/utils';
+import {
+	recordAction,
+	recordGaEvent,
+	recordTrackForPost,
+	recordPermalinkClick
+ } from 'reader/stats';
+
+class PostByline extends React.Component {
+
+	constructor( ) {
+		super( );
+		this.recordTagClick = this.recordTagClick.bind( this );
+		this.recordAuthorClick = this.recordAuthorClick.bind( this );
```

Comment:
React dropped autobinding as part of the move to ES6 classes.  The `onClick` event will find `this.recordTagClick` but within the function call `this.props` will be undefined. 

It's possible to do the binding inside the JSX but there is a [performance hit](https://github.com/yannickcr/eslint-plugin-react/issues/473)  doing that. 

It's possible to sweeten the syntax with:
```
constructor() {
  ...
  this._recordTagClick = () => this.recordTagClick;
  ...
}
```
But I prefer the explicit bind calls.

Looks like es7 has some object property scoping goodness but we're not using the es7 babel preset